src/code/audio/audio_seqplayer.c

```diff
@@ -1,73 +1,2314 @@
+            AudioPlayback_SeqLayerNoteDecay(layer);
+        }
+    }
+}
+
+void AudioSeq_SeqLayerProcessScriptStep1(SequenceLayer* layer) {
+    if (!layer->continuousNotes) {
+        AudioPlayback_SeqLayerNoteDecay(layer);
+    } else if ((layer->note != NULL) && (layer->note->playbackState.wantedParentLayer == layer)) {
+        AudioPlayback_SeqLayerNoteDecay(layer);
+    }
+
+    if ((PORTAMENTO_MODE(layer->portamento) == PORTAMENTO_MODE_1) ||
+        (PORTAMENTO_MODE(layer->portamento) == PORTAMENTO_MODE_2)) {
+        layer->portamento.mode = PORTAMENTO_MODE_OFF;
+    }
+
+    layer->notePropertiesNeedInit = true;
+}
+
+s32 AudioSeq_SeqLayerProcessScriptStep5(SequenceLayer* layer, s32 sameSound) {
+    Note* note;
+
+    if ((layer->continuousNotes == true) && (layer->bit1 == true)) {
+        return 0;
+    }
+
+    if ((layer->continuousNotes == true) && (layer->note != NULL) && layer->bit3 && (sameSound == true) &&
+        (layer->note->playbackState.parentLayer == layer)) {
+        if (layer->sound == NULL) {
+            AudioPlayback_InitSyntheticWave(layer->note, layer);
+        }
+    } else {
+        if (!sameSound) {
+            AudioPlayback_SeqLayerNoteDecay(layer);
+        }
+
+        layer->note = AudioPlayback_AllocNote(layer);
+
+        if (layer->note != NULL) {
+            note = layer->note;
+
+            if (note->playbackState.parentLayer == layer) {
+                AudioEffects_NoteVibratoInit(note);
+            }
+        }
+    }
+
+    if ((layer->note != NULL) && (layer->note->playbackState.parentLayer == layer)) {
+        note = layer->note;
+
+        AudioEffects_NotePortamentoInit(note);
+    }
+
+    return 0;
+}
+
+s32 AudioSeq_SeqLayerProcessScriptStep2(SequenceLayer* layer) {
+    SequenceChannel* channel = layer->channel;
+    SeqScriptState* state = &layer->scriptState;
+    SequencePlayer* seqPlayer = channel->seqPlayer;
+    u8 cmd;
+    u8 cmdArg8;
+    u16 cmdArg16;
+    u16 velocity;
+
+    if (D_801E05B4) {}
+
+    for (;;) {
+        cmd = AudioSeq_ScriptReadU8(state);
+
+        // Note Commands
+        // To be processed in AudioSeq_SeqLayerProcessScriptStep3
+        if (cmd <= 0xC0) {
+            return cmd;
+        }
+
+        // Control Flow Commands
+        if (cmd >= 0xF2) {
+            cmdArg16 = AudioSeq_GetScriptControlFlowArgument(state, cmd);
+
+            if (AudioSeq_HandleScriptFlowControl(seqPlayer, state, cmd, cmdArg16) == 0) {
+                continue;
+            }
+            AudioSeq_SeqLayerDisable(layer);
+            return -1;
+        }
+
+        switch (cmd) {
+            case 0xC1: // layer: set short note velocity
+            case 0xCA: // layer: set pan
+                cmdArg8 = *(state->pc++);
+
+                if (cmd == 0xC1) {
+                    layer->velocitySquare = SQ(cmdArg8) / SQ(127.0f);
+                } else {
+                    layer->pan = cmdArg8;
+                }
+                break;
```

Comment:
Why.... they aren't even the same type of command, why would they jumble them here.